src/Network/ABCI.hs

```diff
@@ -21,74 +22,47 @@ import           Network.ABCI.Types.App               (App (..))
 import qualified Network.ABCI.Types.Error             as Error
 import qualified Network.ABCI.Types.Messages.Request  as Request
 import qualified Network.ABCI.Types.Messages.Response as Response
-import           Network.Socket                       (SockAddr)
 import qualified Proto.Types                          as PT
 import qualified Proto.Types_Fields                   as PT
-import           UnliftIO                             (MonadUnliftIO)
-
 
 -- | Default ABCI app network settings.
 defaultSettings :: ServerSettings
 defaultSettings = serverSettings 26658 $ fromString "127.0.0.1"
 
+type AppRunner m = BS.ByteString -> m BS.ByteString
+
 -- | Serve an ABCI application with custom 'ServerSettings'
 serveAppWith
-  :: ( MonadIO m
-     , MonadBaseControl IO m
-     , MonadUnliftIO m
-     )
-  => ServerSettings
-  -> (SockAddr -> m (App m))
-  -> m ()
-serveAppWith cfg mkApp = runGeneralTCPServer cfg $ \appData -> do
-  app <- mkApp (appSockAddr appData)
-  runConduit (setupConduit app appData)
+  :: ServerSettings
+  -> (AppData -> IO (AppRunner IO))
+  -> IO ()
+serveAppWith cfg mkRunner = runTCPServer cfg $ \appData -> do
+  runner <- mkRunner appData
+  runConduit $
+      appSource appData
+    .| CL.mapM runner
```

Comment:
We just use one mapM now as all other transformations were bunch of maps and one mapM and they can be fused together.